jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+from functools import partial
+from typing import Optional, List, Tuple, Iterable, Any
+
+import scipy
+
+from jax import jit, vmap
+import jax.numpy as jnp
+from jax._src.numpy.util import _wraps
+from jax._src.numpy.lax_numpy import _check_arraylike
+
+ModeResult = namedtuple("ModeResult", ("mode", "count"))
+
+
+@partial(jit, static_argnums=(1,))
+def _mode(x: Iterable[Any], axis: Optional[int]) -> ModeResult:
```

Comment:
I don't think `Iterable[Any]` is the correct annotation here - for example, you use`x.size` attribute below, which is not in general present on an iterable.

Should it be `x: jnp.ndarray`?

Reply:
Eg; ```from numpy.typing import ArrayLike```
https://numpy.org/devdocs/reference/typing.html

You're right, it's not quite right. It's probably semantics, but I felt typing as `jnp.ndarray` limiting given this function should handle anything cleared by the `_check_arraylike` func.

I'll go with `jnp.ndarray` for now, but it may be a point to define exactly in typing form what `_check_arraylike` allows. Not sure if there's enough usecase for a typing module for Jax.

Reply:
And it will definitely be `jnp.ndarray` based on one of your next points, for the hidden functions.